src/Sylius/Bundle/ResourceBundle/spec/Form/EventSubscriber/AddCodeFormSubscriberSpec.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace spec\Sylius\Bundle\ResourceBundle\Form\EventSubscriber;
+
+use PhpSpec\ObjectBehavior;
+use Prophecy\Argument;
+use Sylius\Component\Shipping\Model\ShipmentInterface;
+use Sylius\Component\Shipping\Model\ShippingCategoryInterface;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
+use Symfony\Component\Form\FormInterface;
+
+/**
+ * @author Anna Walasek <anna.walasek@lakion.com>
+ */
+class AddCodeFormSubscriberSpec extends ObjectBehavior
+{
+    function it_is_initializable()
+    {
+        $this->shouldHaveType('Sylius\Bundle\ResourceBundle\Form\EventSubscriber\AddCodeFormSubscriber');
+    }
+
+    function it_subscribes_to_event()
+    {
+        $this::getSubscribedEvents()->shouldReturn(array(FormEvents::PRE_SET_DATA => 'preSetData'));
+    }
+
+    function it_sets_code_as_enabled(FormEvent $event, FormInterface $form, ShippingCategoryInterface $shippingCategory)
```

Comment:
``it_sets_code_as_enabled_when_resource_is_new``

Also, we should use ``ResourceInterface`` here, otherwise we are coupling this spec to shipping component. I know that ResourceInterface is not here yet, but commenting to keep this in mind after we merge #3617.